[DOCS] Update patching options and daal4py install guide (#880)
intel/scikit-learn-intelex: doc/sources/what-is-patching.rst

         python sklearnex.glob patch_sklearn
      
      .. seealso:: :ref:`global_patching`
   

>    how it works

Reviewer: this is the syntax to introduce a new term in a glossary, meaning that you are introducing a "how it works" term, I don't think this is a good approach

you can make it a subsection, e.g.:

```
How patching works
-----------------------
```

or: introduce a definition of `dispatching` (the second sentence of the definition below), but the "how it works" still shouldn't be in the glossary

>       We are replacing the original estimators in scikit-learn modules with ours.
>       There is dispatching inside our estimators, if the desired algorithm parameters
>       are not supported by the |intelex|, then the result of the original scikit-learn
>       is returned.

Reviewer: better to remove we/ours, and the second sentence is too long and messy

`There is dispatching inside our estimators` -- I don't really understand what you mean by this part (I mean, I know what you mean, but this particular phrase is confusing). I think the message is clear even without it (the if-then part of the sentence is basically explaining this). See the comment above about introducing the `dispatching` term to explain what happens when the parameter is not supported by intelex

```suggestion
      The extension replaces the original estimators in scikit-learn modules with the optimized ones.
      There is dispatching inside our estimators. If the desired algorithm parameters
      are not supported by |intelex|, the result of the original scikit-learn
      is returned.
```
